src/main/java/org/broadinstitute/barclay/help/WDLWorkUnitHandler.java

```diff
@@ -0,0 +1,337 @@
+     * @param argDef the Barclay NamedArgumentDefinition for this arg
+     * @param argDocType the display type as chosen by the Barclay doc system for this arg. this is what
+     * @return
+     */
+    protected String getWDLTypeForArgument(
+            final ArgumentDefinition argDef,
+            final String argDocType
+    ) {
+        final Field argField = argDef.getUnderlyingField();
+        final Class<?> argumentClass = argField.getType();
+
+        // start the data type chose by the doc system and transform that based on the underlying
+        // java class/type
+        String wdlType = argDocType;
+
+        // if the underlying field is a collection type; it needs to map to "Array", and then the
+        // type param has to be converted to a WDL type
+        if (argDef.isCollection()) {
+            Pair<String, String> conversionPair = tranformToWDLCollectionType(argumentClass);
+            if (conversionPair != null) {
+                wdlType = wdlType.replace(conversionPair.getLeft(), conversionPair.getRight());
+            } else {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Unrecognized collection type %s for argument %s in work unit %s." +
+                                "Argument collection type must be one of List or Set.",
+                                argumentClass,
+                                argField.getName(),
+                                argField.getDeclaringClass()));
+            }
+
+            // Convert any Collection type params; this only handles a single generic type parameter (i.e List<T>),
+            // where type T can in turn be a generic type, again with a single type param. This is sufficient to
+            // accommodate all existing cases, with the most complex being List<FeatureInput<T>>. If this code
+            // encounters anything more complex, it will throw. Supporting more deeply nested generic types will
+            // require additional code.
+            final Type typeParamType = argField.getGenericType();
+            if (typeParamType instanceof ParameterizedType) {
+                final ParameterizedType pType = (ParameterizedType) typeParamType;
+                final Type genericTypes[] = pType.getActualTypeArguments();
+                if (genericTypes.length != 1) {
+                    throw new RuntimeException(String.format(
+                            "Generating WDL for tools with arguments that have types that require multiple type parameters is not supported " +
+                                    "(class %s for arg %s in %s has multiple type parameters).",
+                            argumentClass,
+                            argField.getName(),
+                            argField.getDeclaringClass()));
+                }
+                ParameterizedType pType2 = null;
+                Type genericTypes2[];
+                Class<?> nestedTypeClass;
```

Comment:
Will this handle nested collections?  Like a `List<List<FeatureInput<VariantContext>>>`?  Not sure if we have any right now, but we may at some point

Reply:
No. It will handle 2 levels of nesting (sufficient for `List<FeatureInput<VariantContext>>`, which is as deep as anything currently in GATK). If at some point we need more than that we'll have to expand as needed.

Reply:
BTW, this specific example you cited (`List<List<FeatureInput<VariantContext>>>`) isn't really supported by the argument parser because there is no way for the user to group the inner list members on the command line.

Reply:
Sounds good.

Ah.  Interesting.  I guess we don't need anything like that right now anyway.